packages/ai-jsx/src/batteries/constrained-output.tsx

```diff
@@ -5,169 +5,255 @@
  */
 
 import * as AI from '../index.js';
-import { ChatCompletion, SystemMessage, AssistantMessage, UserMessage } from '../core/completion.js';
+import {
+  ChatCompletion,
+  SystemMessage,
+  AssistantMessage,
+  UserMessage,
+  ModelPropsWithChildren,
+} from '../core/completion.js';
 import yaml from 'js-yaml';
 import { AIJSXError, ErrorCode } from '../core/errors.js';
+import z from 'zod';
+import { zodToJsonSchema } from 'zod-to-json-schema';
+import untruncateJson from 'untruncate-json';
 
-interface ValidationResult {
-  success: boolean;
-  error: string;
-}
+export type ObjectCompletion = ModelPropsWithChildren & {
+  /** Validators are used to ensure that the final object looks as expected. */
+  validators?: ((obj: object) => void)[];
+  /**
+   * The object Schema that is required. This is a type of validator with the excpetion
+   * that the schema description is also provided to the model.
+   */
+  schema?: z.Schema;
```

Comment:
Seems like the doc strings are not showing up in the docs site. Should I be using `interface` instead? But `interface` didn't let me easily do `&` and other stuff. I didn't try too much either tbh :)

Reply:
> Seems like the doc strings are not showing up in the docs site.

You mean it's not showing up in the generated markdown docs? Hmm. It seems like it should, based on what you're doing.